jwst/ramp_fitting/ramp_fit.py

```diff
@@ -474,69 +498,84 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
     #     all integrations:
     #     slope = sum_over_integs_and_segs(slope_seg/var_seg)/
     #                    sum_over_integs_and_segs(1/var_seg)
+    slope_by_var4 = opt_res.slope_seg.copy()/var_both4 
 
-    slope_seg = opt_res.slope_seg.copy()
-    slope_by_var4 = slope_seg / var_both4
-    slope_by_var4 = utils.rem_nan_inf(slope_by_var4)
+    del var_both4
 
-    s_slope_by_var3 = slope_by_var4.sum(axis=1) # sum over SEGMENTS (not integs)
+    s_slope_by_var3 = slope_by_var4.sum(axis=1) # sum over segments (not integs)
     s_slope_by_var2 = s_slope_by_var3.sum(axis=0) # sum over integrations
-
     s_inv_var_both2 = s_inv_var_both3.sum(axis=0)
 
     # Compute the 'dataset-averaged' slope
     slope_dataset2 = s_slope_by_var2/s_inv_var_both2
-    slope_dataset2 = utils.rem_nan_inf(slope_dataset2)
+
+    del s_inv_var_both2, s_slope_by_var2, s_slope_by_var3, slope_by_var4
+    del s_inv_var_both3
+
+    #  Replace nans in slope_dataset2 with 0 (for non-existing segments)
+    slope_dataset2[np.isnan(slope_dataset2)] = 0.
 
     # Compute the integration-specific slope
-    the_num = (slope_seg * inv_var_both4).sum(axis = 1) # sum over all segments
+    the_num = (opt_res.slope_seg.copy() * inv_var_both4).sum(axis = 1) 
```

Comment:
I don't think you need to copy `opt_res.slope_seg`, since it's used in an expression rather than being assigned directly.